app/presenters/user_list_presenter.rb

```diff
@@ -2,14 +2,14 @@ class UserListPresenter
   attr_reader :type, :language, :location
   
   def initialize(params)
-    @type = sanitize_type(type: params[:type])
+    @type = (params.keys.map(&:to_sym) & [:city, :country]).first || :world
     @page = [params[:page].to_i, 1].max
-    @location = params[@type].try(:downcase).try(:strip) || default_location
+    @location = params.with_indifferent_access[@type].try(:downcase).try(:strip)
     @language = params[:language] || "JavaScript"
```

Comment:
What do you think about using the absolute ranking for all languages rather than JavaScript language?